src/hamiltonian.jl

```diff
@@ -513,6 +514,8 @@ _similarmatrix(::Type{A}, ::Type{A´}, h) where {N,T<:SMatrix{N,N},A<:Matrix{T},
     similar(A, size(h))
 _similarmatrix(::Type{A}, ::Type{A´}, h) where {N,T<:Number,A<:Matrix{T},T´<:SMatrix{N,N},A´<:AbstractMatrix{T´}} =
     similar(A, flatsize(h))
+_similarmatrix(::Type{A}, ::Type{A´}, ketflatsize::Tuple) where {N,T<:Number,A<:Matrix{T},T´<:SMatrix{N,N},A´<:AbstractMatrix{T´}} =
```

Comment:
Better to not add a _similarmatrix method with a different signature. Note that the type of `h` in the other `_similarmatrix` method is not type-constrained, so it could accept a `ket::AbstractMatrix`